subprojects/core/src/main/java/org/gradle/util/internal/NameMatcher.java

```diff
@@ -63,6 +63,7 @@ public <T> T find(String pattern, Map<String, ? extends T> items) {
      *
      * @return The match if exactly 1 match found, null if no matches or multiple matches.
      */
+    @SuppressWarnings("deprecation")
```

Comment:
Can we put this suppression on the line it is used so we don't accidentally mask other deprecations?